diff -Naur a/frameworks/base/core/java/android/app/admin/DevicePolicyManager.java b/frameworks/base/core/java/android/app/admin/DevicePolicyManager.java
--- a/frameworks/base/core/java/android/app/admin/DevicePolicyManager.java	2019-11-22 23:08:39.629191194 +1100
+++ b/frameworks/base/core/java/android/app/admin/DevicePolicyManager.java	2019-11-22 23:09:46.146330832 +1100
@@ -3127,7 +3127,7 @@
             return 0;
         }
         // Kind-of arbitrary.
-        return 16;
+        return 64;
     }
     /**
      * Determines whether the calling user's current password meets policy requirements
